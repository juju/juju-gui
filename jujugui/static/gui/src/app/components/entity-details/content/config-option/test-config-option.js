/* Copyright (C) 2017 Canonical Ltd. */
'use strict';

const React = require('react');
const enzyme = require('enzyme');

const EntityContentConfigOption = require('./config-option');

describe('EntityContentConfigOption', function() {
  const renderComponent = (options = {}) =>
    enzyme.shallow(<EntityContentConfigOption option={options.option} />);

  it('can render with a default value', function() {
    var option = {
      name: 'password',
      description: 'Required password',
      type: 'string',
      default: 'abc123'
    };
    const wrapper = renderComponent({option});
    expect(wrapper).toMatchSnapshot();
  });

  it('can render without a default value', function() {
    var option = {
      name: 'password',
      description: 'Required password',
      type: 'string'
    };
    const wrapper = renderComponent({option});
    assert.equal(wrapper.find('.entity-content__config-default').length, 0);
  });
});
