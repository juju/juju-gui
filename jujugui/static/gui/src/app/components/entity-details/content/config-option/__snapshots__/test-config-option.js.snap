// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityContentConfigOption can render with a default value 1`] = `
<div
  className="entity-content__config-option"
>
  <dt
    className="entity-content__config-name"
    id="charm-config-password"
  >
    password
  </dt>
  <dd
    className="entity-content__config-description"
  >
    <p>
      <span
        className="entity-content__config-type"
      >
        (
        string
        )
      </span>
       
      Required password
    </p>
  </dd>
  <dd
    className="entity-content__config-default"
  >
    abc123
  </dd>
</div>
`;
